02_bubble/n0npax/main_test.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"bytes"
+	"math/rand"
+	"sort"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("[1 2 3 5]\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
+	}
+}
+
+func randomSlice(size int) []int {
+	s := make([]int, size)
+	for i := 0; i < size; i++ {
+		s[i] = rand.Intn(100) - 50
+	}
+	return s
+}
+
+type fn func([]int) []int
+
+func TestBubble(t *testing.T) {
+	for size := 0; size < 10; size++ {
+		testFunction(t, bubble, "bubble", size)
+	}
+}
+
+func TestInsertion(t *testing.T) {
+	for size := 0; size < 10; size++ {
+		testFunction(t, insertion, "insertion", size)
+	}
+}
+
+func TestQuicksort(t *testing.T) {
+	for size := 0; size < 10; size++ {
+		testFunction(t, quicksortWrapper, "quicksort", size)
+	}
+}
+
+func testFunction(t *testing.T, f fn, name string, size int) {
```

Comment:
could be `TestSort` with `testCases map[string]fn{ "bubble": bubble, "quick": quicksortWrapper/*...*/ }` than loop over the different sort implementations.

Reply:
Thanks for that! Fixed